javascript/ql/src/Security/CWE-022/ZipSlip.ql

```diff
@@ -0,0 +1,22 @@
+/**
+ * @name Arbitrary file write during zip extraction ("Zip Slip")
+ * @description Extracting files from a malicious zip archive without validating that the
+ *              destination file path is within the destination directory can cause files outside
+ *              the destination directory to be overwritten.
+ * @kind path-problem
+ * @id js/zipslip
+ * @problem.severity error
+ * @precision high
```

Comment:
Let's start this out as `medium` precision. That way it'll be run on LGTM, but results won't be displayed by default yet. After the next dist upgrade, we can take a look at the results and then decide whether we want to display by default or not.

Reply:
Sounds good; thanks for pointing out that side-effect, which caused me to find the wiki page documenting the whole precision x severity table.

Reply:
Do you mind if I postpone the `tarslip` case to another PR, so comments are less tangled? I do have the commit sitting in my back pocket  on a local branch so I can just rebase it.

Reply:
I am interested in performance profiling --- this is using `dist-compare`, right?

Reply:
Sure. (Btw, we don't generally insist on squashing PRs down to a single commit if that's what you're worried about.)

Reply:
> I am interested in performance profiling --- this is using `dist-compare`, right?

Correct. I'll ping you on slack about this.